packages/devextreme/js/__internal/core/utils/m_date.ts

```diff
@@ -234,7 +234,11 @@ function addDateInterval(value, interval, dir) {
     result.setMonth(result.getMonth() + 3 * intervalObject.quarters * dir);
   }
   if (intervalObject.months) {
-    result.setMonth(result.getMonth() + intervalObject.months * dir);
+    const targetMonth = result.getMonth() + intervalObject.months * dir;
```

Comment:
[test coverage]
If u change core util, u should write a test for this util

Reply:
[7458d66](https://github.com/DevExpress/DevExtreme/pull/28423/commits/7458d66f3dcb1882806a361d8f83596b6bf1ef54)